### PR TITLE
cloudapi: rework return values

### LINKED ISSUE #1606: cloudapi: Replace `panic`s with 500s
There's still several places where the handlers will panic instead of returning an InternalServerError. Calling a handler shouldn't crash composer.

### PR DESCRIPTION
**Pull request by chloenayon**
Change instances of panic on error to return StatusInternalServerError
instead. Additionally change return code 200s to http.StatusOK.

Fixes #1606

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Review by ondrejbudai — APPROVED** [review]
Look good, thanks!

**Review by thozza — DISMISSED** [review]
Thanks for the fix! Changes look good to me.

Nitpick: It would be probably better to be consistent in how we format the message in `echo.NewHTTPError()` calls with the rest of the code. From a brief look, there seems to be one instance when we use `fmt.Snprintf()`:
https://github.com/osbuild/osbuild-composer/blob/a3ac31a4830604cf0d25f91236e092512cddab13/internal/cloudapi/server.go#L81

while in all other relevant cases we simply pass the arguments for the message format string as additional values:
https://github.com/osbuild/osbuild-composer/blob/a3ac31a4830604cf0d25f91236e092512cddab13/internal/cloudapi/server.go#L146
https://github.com/osbuild/osbuild-composer/blob/a3ac31a4830604cf0d25f91236e092512cddab13/internal/cloudapi/server.go#L180
https://github.com/osbuild/osbuild-composer/blob/a3ac31a4830604cf0d25f91236e092512cddab13/internal/cloudapi/server.go#L184
https://github.com/osbuild/osbuild-composer/blob/a3ac31a4830604cf0d25f91236e092512cddab13/internal/cloudapi/server.go#L216
...

My suggestion would be to stick to the dominant style - pass arguments of the message as additional values when calling `echo.NewHTTPError()`, instead of string concatenation. It would be also great if you could change https://github.com/osbuild/osbuild-composer/blob/a3ac31a4830604cf0d25f91236e092512cddab13/internal/cloudapi/server.go#L81 to the same style to make the whole file consistent.

File: internal/cloudapi/server.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (b binder) Bind(i interface{}, ctx echo.Context) error {
 
 	err := json.NewDecoder(ctx.Request().Body).Decode(i)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Cannot parse request body: %v", err))
+		return echo.NewHTTPError(http.StatusBadRequest, "Cannot parse request body: %s", err)
```
Comment (thozza) [inline internal/cloudapi/server.go:81]: You should use `err.Error()` instead of `err` here, to make sure that the passed value type is `string`.

**Review by thozza — DISMISSED** [review]
LGTM

**Review by thozza — APPROVED** [review]
Still good!

**Pull request merged**